Index: ltq-vrx518-2018-12-06-0a301512/aca.c
===================================================================
--- ltq-vrx518-2018-12-06-0a301512.orig/aca.c
+++ ltq-vrx518-2018-12-06-0a301512/aca.c
@@ -155,7 +155,7 @@ static const struct firmware *aca_fetch_
 	char filename[64];
 	const struct firmware *fw;
 
-	snprintf(filename, sizeof(filename), "%s/%s", CONFIG_VRX518_FW_PATH,
+	snprintf(filename, sizeof(filename), "%s/%s", "/lib/firmware/vrx518",
 		ACA_FW_FILE);
 	ret = request_firmware(&fw, filename, priv->dev);
 	if (ret) {
--- ltq-vrx518-2018-12-06-0a301512.orig/tc/tc_api.c
+++ ltq-vrx518-2018-12-06-0a301512/tc/tc_api.c
@@ -1057,7 +1057,7 @@ int ppe_fw_load(struct tc_priv *priv)
 	u32 *dst;
 	char fw_name[48];
 
-	snprintf(fw_name, sizeof(fw_name), "%s/%s", CONFIG_VRX518_FW_PATH,
+	snprintf(fw_name, sizeof(fw_name), "%s/%s", "/lib/firmware/vrx518",
 		ppe_fw_name);
 	if (request_firmware(&priv->fw.fw, fw_name, priv->ep_dev[0].dev)) {
 		tc_err(priv, MSG_INIT,
