--- b/src/linux/ifxos_linux_socket_drv.c	2016-09-27 16:31:31.000000000 +0200
+++ b/src/linux/ifxos_linux_socket_drv_mod.c	2020-02-04 02:04:37.637543492 +0100
@@ -191,8 +191,11 @@
       kernel space, otherwise sock_recvmsg() will fail.*/
    old_fs = get_fs();
    set_fs(KERNEL_DS);
-
+#if LINUX_VERSION_CODE < KERNEL_VERSION(4,7,0)
    ret = sock_recvmsg ((struct socket *) socFd, &msg, bufSize_byte, 0);
+#else
+   ret = sock_recvmsg ((struct socket *) socFd, &msg, 0);
+#endif
    set_fs(old_fs);
 
    return ret;
